client/core/core.go

```diff
@@ -4606,6 +4653,55 @@ func (c *Core) bondKeysReady() bool {
 	return c.bondXPriv != nil && c.bondXPriv.IsPrivate() // infer not Zeroed via IsPrivate
 }
 
+// encodeRecoverySeed encrypts the raw seed and app password using part of the
+// original seed.
+func (c *Core) encodeRecoverySeed(rawSeed, pw []byte) ([]byte, []byte, error) {
+	var encRecoverySeedPw = rawSeed[:seedLen/2]
+
+	encRecoverySeedCrypter := c.newCrypter(encRecoverySeedPw)
+	defer encRecoverySeedCrypter.Close()
+
+	seedAndPass := append(rawSeed, pw...)
+	encRecoverySeed, err := encRecoverySeedCrypter.Encrypt(seedAndPass)
+	if err != nil {
+		return nil, nil, fmt.Errorf("recovery seed encryption error: %v", err)
+	}
+
+	return encRecoverySeed, encRecoverySeedCrypter.Serialize(), nil
+}
+
+// checkIfSeedMatch checks if the provided seed matches the stored recovery seed
+// and returns the decrypted app password.
+func (c *Core) checkIfSeedMatch(seed []byte) ([]byte, error) {
+	creds := c.creds()
+	if creds == nil {
+		return nil, fmt.Errorf("no credentials stored")
+	}
+
+	if len(seed) != seedLen {
+		return nil, fmt.Errorf("invalid seed length %d", len(seed))
+	}
+
+	var encRecoverySeedPW = seed[:seedLen/2]
+	encRecoverySeedCrypter, err := c.reCrypter(encRecoverySeedPW, creds.EncRecoverySeedParams)
```

Comment:
This effectively reduces our seed to a 32-byte seed. Before they had to guess our seed out of 2^512 possible seeds. Now they just have to guess half the seed, so 2^256.

Reply:
Yh, do you think using the whole seed is better? 🌚  I felt we needed just some part of it.

Reply:
It'll only be as safe as guessing a sha256 preimage now.

Reply:
@buck54321, do you have any suggestions in regards to this?